examples/run/openmp/par_example.cpp

```diff
@@ -44,105 +43,52 @@ int par_run(const std::string &detector_file, const std::string &cells_dir,
-                cc(cells_per_event.at(i).items, cells_per_event.at(i).header);
-            for (auto &cl_id : clusters.get_headers()) {
-                cl_id.pixel = module.pixel;
-            }
-
-            traccc::host_measurement_collection measurements_per_module =
-                mt(clusters, module);
-            traccc::host_spacepoint_collection spacepoints_per_module =
-                sp(module, measurements_per_module);
-            // The algorithmnic code part: end
-
-            n_cells += cells_per_event.at(i).items.size();
-            n_clusters += clusters.size();
-            n_measurements += measurements_per_module.size();
-            n_space_points += spacepoints_per_module.size();
-
-#pragma omp critical
-            {
-                measurements_per_event.push_back(
-                    module, std::move(measurements_per_module));
-
-                spacepoints_per_event.push_back(
-                    module.module, std::move(spacepoints_per_module));
-            }
-        }
-
-        traccc::measurement_writer mwriter{
-            traccc::get_event_filename(event, "-measurements.csv")};
-        for (size_t i = 0; i < measurements_per_event.size(); ++i) {
-            auto measurements_per_module = measurements_per_event.at(i).items;
-            auto module = measurements_per_event.at(i).header;
-            for (const auto &measurement : measurements_per_module) {
-                const auto &local = measurement.local;
-                mwriter.append({module.module, "", local[0], local[1], 0., 0.,
-                                0., 0., 0., 0., 0., 0.});
-            }
-        }
-
-        traccc::spacepoint_writer spwriter{
-            traccc::get_event_filename(event, "-spacepoints.csv")};
-        for (size_t i = 0; i < spacepoints_per_event.size(); ++i) {
-            auto spacepoints_per_module = spacepoints_per_event.at(i).items;
-            auto module = spacepoints_per_event.at(i).header;
-
-            for (const auto &spacepoint : spacepoints_per_module) {
-                const auto &pos = spacepoint.global;
-                spwriter.append({module, pos[0], pos[1], pos[2]});
-            }
-        }
+            traccc::read_cells_from_event(event, cells_dir, surface_transforms,
+                                          resource);
+
+        /*-------------------
+            Clusterization
+          -------------------*/
+
+        auto measurements_per_event = ca(cells_per_event);
+
+        /*------------------------
+            Spacepoint formation
+          ------------------------*/
+
+        auto spacepoints_per_event = sf(measurements_per_event);
+
+        /*----------------------------
+          Statistics
+          ----------------------------*/
+
+        n_modules += cells_per_event.size();
+        n_cells += cells_per_event.total_size();
+        n_measurements += measurements_per_event.total_size();
+        n_spacepoints += spacepoints_per_event.total_size();
     }
```

Comment:
Here it should be a critical region to aggregate the measurements and spacepoints.

Reply:
Done